dms/models/dms_file.py

```diff
@@ -617,6 +644,35 @@ def _inverse_content(self):
             for vals, ids in updates.items():
                 self.browse(ids).write(dict(vals))
 
+    def _create_model_attachment(self, vals):
+        res_vals = vals.copy()
+        if self.env.context.get("default_directory_id"):
```

Comment:
This is still using the infamous context key already commented about. Why not using `vals["directory_id"]`? And why you need to browse and then get only the ID?

Reply:
Honestly i don't understand because `default_directory_id` is needed but now is used in some places:

https://github.com/OCA/dms/blob/12.0/dms/models/res_company.py#L109
https://github.com/OCA/dms/blob/12.0/dms/static/src/js/views/many_drop_target.js#L83
https://github.com/OCA/dms/blob/12.0/dms/views/directory.xml#L45

Try to change it and i search a better way?

Reply:
As said, that's a context value that serves for having default values populated, but you mustn't relay on it, but on vals dictionary, as if you programatically create a record, you must provide 2 times same data. Do you want to meet for more explanations?